src/components/duelling-picklist/duelling-picklist.pw.tsx

```diff
@@ -0,0 +1,743 @@
+      await groupAddButton.click();
+      await expect(
+        getDataElementByValue(page, "picklist")
+          .nth(0)
+          .locator("li")
+          .locator("p")
+      ).toHaveCount(3);
+      await expect(
+        getDataElementByValue(page, "picklist")
+          .nth(1)
+          .locator("li")
+          .locator("p")
+      ).toHaveCount(3);
+      const groupRemoveButton = getDataElementByValue(page, "picklist-group")
+        .nth(2)
+        .locator("button");
+      await groupRemoveButton.click();
+      await expect(
+        getDataElementByValue(page, "picklist")
+          .nth(0)
+          .locator("li")
+          .locator("p")
+      ).toHaveCount(6);
+      await expect(
+        getDataElementByValue(page, "picklist")
+          .nth(1)
+          .locator("li")
+          .locator("p")
+      ).toHaveCount(0);
+    });
+  }
+);
+
+test.describe(`check events for Duelling-Picklist component`, () => {
+  test(`should call onChange when add button clicked`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DuellingPicklistComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const addItemButton = page.getByRole("button").first();
+    await addItemButton.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onChange when remove button clicked`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DuellingPicklistComponentAssigned
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const removeItemButton = page.getByRole("button").first();
+    await removeItemButton.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  [...keyToTrigger].forEach((pressed) => {
+    test(`should call onChange when ${pressed} key pressed on add button`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <DuellingPicklistComponentAssigned
+          onChange={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const addItemButton = page.getByRole("button").first();
+      await addItemButton.press(pressed);
+      expect(callbackCount).toBe(1);
+    });
+  });
+
+  [...keyToTrigger].forEach((pressed) => {
+    test(`should call onChange when ${pressed} key pressed on remove button`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <DuellingPicklistComponentAssigned
+          onChange={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const removeItemButton = page.getByRole("button").first();
+      await removeItemButton.press(pressed);
+      expect(callbackCount).toBe(1);
+    });
+  });
+});
+
+test.describe(`Accessibility tests for Duelling-Picklist component`, () => {
+  test(`should pass accessibility tests for default story`, async ({
```

Comment:
**suggestion(non-blocking):** As these aren't technically stories anymore, this should be changed to something like this or something better. If You or the QA reviewer can be more creative with a description, go for it.

```suggestion
  test(`should pass accessibility tests for default example`, async ({
```